### ISSUE TITLE
Build failed on 2.8 dev due to new HttpHeaders.set method argument

### ISSUE THREAD

**Issue by alextekartik** [pulyaevskiy/node-interop#74]
I don't know if you have CI running on dev. The projects I have failed to compile on dev (2.8.0-dev.10)
```
00:02 +0: [Node.js] compiling test/context_node_test.dart                                                                                                                                              
../../../../.pub-cache/hosted/pub.dartlang.org/node_io-1.0.1+2/lib/src/http_headers.dart:314:8:
Error: The method 'HttpHeaders.set' has fewer named arguments than those of overridden method 'HttpHeaders.set'.
  void set(String name, Object value) {
```

Example here: https://travis-ci.org/tekartik/platform.dart/builds/653803172

There is a new argument `preserveHeaderCase` for HttpHeaders.set
https://api.dart.dev/dev/2.8.0-dev.10.0/dart-io/HttpHeaders/set.html
that your class should add.

**Comment by whesse** [comment]
I have hit this too, and fixed it locally by just adding the new optional argument
, {bool preserveHeaderCase: false}
after String name, Object value
in the functions add() and set(), lines 273 and 314 of node_io/lib/src/http_headers.dart.

Unless there is an equivalent node parameter to "preserveHeaderCase", I wouldn't implement anything based on the new parameter.

**Comment by whesse** [comment]
I have started working on a pull request to fix this, but there are additional complexities:

Because this is a breaking change, the sdk constraint in the pubspec.yaml should be changed to
```
environment:
  sdk: '>=2.8.0-dev.10.0 <3.0.0'
```

This may be significant enough that the version of this package should be changed to 1.0.2 or 1.1.0.
It may be desirable to publish a final version without the fix, with sdk constrained to <2.8.0-dev.10.0,
to indicate that that version is not compatible with newer SDKs.  I am not sure of best practice for this situation.

When the sdk version is constrained to 2.8 or later, the backward-compatible fixes introduced in #61 and perhaps #59 #60 #63 should be simplified as suggested in those PRs, since they no longer need to be backward compatible.

To summarize, this change should also change the pubspec.yaml, the changelog, the version number, and carry out the recommendations in #61.

**Comment by barapa** [comment]
I tried the solution suggested in https://github.com/pulyaevskiy/node-interop/issues/74#issuecomment-595677222, but have not been successful.

I updated the file `build/packages/node_io/src/http_headers.dart`

But when building via `pub run build_runner build --output=build` still outputs
```
Dart2Js finished with:

packages/node_io/src/http_headers.dart:273:8:
Error: The method 'HttpHeaders.add' has fewer named arguments than those of overridden method 'HttpHeaders.add'.
  void add(String name, Object value) {
       ^
org-dartlang-sdk:///sdk/lib/_http/http.dart@24654+1:
Info: This is the overridden method ('add').
packages/node_io/src/http_headers.dart:314:8:
Error: The method 'HttpHeaders.set' has fewer named arguments than those of overridden method 'HttpHeaders.set'.
  void set(String name, Object value) {
       ^
org-dartlang-sdk:///sdk/lib/_http/http.dart@24903+1:
Info: This is the overridden method ('set').
Error: Compilation failed.
```

**Comment by whesse** [comment]
I believe you would need to update the file http_headers.dart in the source directory the packages are downloaded into, not just the build directory the packages are copied into when building.  You can see from the error message that it is finding a copy of the file where those lines have not been changed.

**Comment by barapa** [comment]
Any update on this? I was able to work around this by migrating flutter from the beta to the stable channel, but that appears to be causing other issues.

**Comment by pulyaevskiy** [comment]
The fix has been released a few days ago. Closing this issue.
